auth/downscoping/token_consumer.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package downscopingoverview
+
+// [START auth_overview_downscoping_token_consumer]
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/oauth2"
+	"google.golang.org/api/option"
+)
+
+// A token consumer should define their own tokenSource. In the Token() method,
+// it should send a query to a token broker requesting a downscoped token.
+// The token broker holds the root credential that is used to generate the
+// downscoped token.
+type localTokenSource struct {
+	requestedObject string
+	brokerURL       string
+}
+
+func (localTokenSource) Token() (*oauth2.Token, error) {
+	var remoteToken oauth2.Token
+	// Retrieve remoteToken, an oauth2.Token, from token broker.
+	return &remoteToken, nil
+}
+
+func getObjectContents(ctx context.Context) ([]byte, error) {
```

Comment:
Please provide doc comment: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#comment-functions-and-packages